packages/devtools_server/lib/src/memory_profile.dart

```diff
@@ -131,54 +131,53 @@ class MemoryProfile {
   // TODO(terry): Investigate moving code from this point through end of class to devtools_shared.
   void startPolling() {
     _pollingTimer = Timer(updateDelay, _pollMemory);
-    service.onGCEvent.listen(_handleGCEvent);
+    service!.onGCEvent.listen(_handleGCEvent);
   }
 
   void _handleGCEvent(Event event) {
     //final bool ignore = event.json['reason'] == 'compact';
-
+    final json = event.json!;
     final List<HeapSpace> heaps = <HeapSpace>[
-      HeapSpace.parse(event.json['new']),
-      HeapSpace.parse(event.json['old'])
+      HeapSpace.parse(json['new'])!,
+      HeapSpace.parse(json['old'])!
     ];
-    _updateGCEvent(event.isolate.id, heaps);
+    _updateGCEvent(event.isolate!.id!, heaps);
     // TODO(terry): expose when GC occured as markers in memory timeline.
   }
 
   void stopPolling() {
-    _pollingTimer?.cancel();
+    _pollingTimer.cancel();
     service = null;
   }
 
   Future<void> _pollMemory() async {
-    final VM vm = await service.getVM();
+    assert(service != null);
```

Comment:
cleanup by writing
final service = !this.service;
optionally: after each async block you could verify that service == this.service and abort otherwise as the service has changed so the memory polling request should be abandoned.

Reply:
changed to `final service = this.service!`